invenio_rdm_records/vocabularies/__init__.py

```diff
@@ -106,37 +107,40 @@ def get_invalid(self, type_subtype):
             )
         )
 
-    def dump_text_values(self):
-        """Returns json-compatible key-part: texts and the values.
+    def dump_options(self):
+        """Returns json-compatible dict of options for type and subtype.
 
         The current shape is influenced by current frontend, but it's flexible
         enough to withstand the test of time (new frontend would be able to
-        change it easily).
+        adapt it to their needs easily).
 
         TODO: Be attentive to generalization for all vocabularies.
         """
-        text_values = {'type': [], 'subtype': []}
+        options = {'type': [], 'subtype': []}
 
         for (_type, subtype), entry in self.data.items():
             type_option = {
+                'icon': entry.get('type_icon'),
```

Comment:
Curiosity: This is for the UI icon right?

Reply:
Yes it's for the UI. As much as possible I want to concentrate a lot of these customizations there so that it's easy to change by everyone. Labels, icons and order have been asked to be customized frequently by librarians in my group.